src/main/java/org/lmdbjava/MaskedFlag.java

```diff
@@ -64,7 +65,9 @@ static int mask(final MaskedFlag... flags) {
    * @return true if set.
    */
   static boolean isSet(final int flags, final MaskedFlag test) {
-    requireNonNull(test);
+    if (SHOULD_CHECK) {
+      requireNonNull(test);
+    }
```

Comment:
Please remove the `SHOULD_CHECK` condition, as it is unlikely to be faster than an inlined `requireNonNull` call. Most of the other `SHOULD_CHECK`s are guarding multiple conditions or expensive checks.